### DISCUSSION TITLE
List of Potential Standardized Utility Functions To Consider or Complete

### DISCUSSION THREAD

**Discussion by gwu05** [Gilead-Public/gsm#14] [General]
Calculation of treatment start and end date for Exposure
-Derive FDD and LDD
-may require EXSTDAT and EXENDAT, DADPDAT, DARETDAT

Identification Parsing
-SUBJID, INVID, SITEID

Study Duration
Want First Day in Study
Want Last Day in Study

Mark Study Completion
-Complete Study (Y or N)
-Reason for not completing study

Mark Treatment Completion
-Complete Trt (Y or N)
-Reason for not completing trt

Measurement Date and Visit
-Assign Analysis Visit (AVISITN)
 -Calculate study day
 -Setup analysis window table
 -Assign Analysis window

Measurement Outcome Info
-Baseline value
-Change
-Percentage change

Select Value for Analysis
-multiple records assigned to same visit window
-for baseline, last nonmissing value on or prior to FDD
-for post-baseline, record closest to nominal day,  or if two records equidistant, later record select

PValue Thresholding
-FDR, bejaminimi hochberg, outlier, etc.

Diagnostic Plot for PValue

Autothresholding for PValues

**Comment by gwu05** [comment]
For TEAE and TE Lab Abnormalities from raw data, there are also specific rules in order to determine those.

For example:
**TEAEs**
Treatment-emergent AEs will be separately defined for the double-blind phase and the open-label phase. Post-treatment-emergent AEs will be defined only for the TFFU phase.
Treatment emergent AEs occurring during the double-blind phase are defined as:
•	Any AE with onset date on or after the blinded study drug start date and no later than the minimum of the blinded study drug stop date + 3 days and the day prior to the first dose date of open-label study drug, if applicable, for those who discontinued blinded study drug permanently, or
•	Any AE with onset date on or after the blinded study drug start date for those who are still on the blinded study drug, or
•	Any AE leading to blinded study drug discontinuation.
Treatment-emergent AEs occurring during the open-label phase are defined as:
•	Any AE with onset date on or after the open-label study drug start date and no later than the open-label study drug stop date + 3 days for those who discontinued open-label study drug permanently, or
•	Any AE with onset date on or after the open-label study drug start date for those who are still on open-label study drug, or
•	Any AE leading to open-label study drug discontinuation.
Post-treatment-emergent AEs occurring during the TFFU phase are defined as:
•	Any AE with onset date after the last dose of study drug + 3 days 
7.1.6.2.	Incomplete Dates
If an AE onset date is incomplete or completely missing, the following rules will be used to determine if the AE is considered treatment emergent during the double-blind and open-label phases, or post-treatment-emergent during the TFFU phase: 
Events with Missing Onset Day and/or Month
The event is treatment-emergent during the double-blind phase if the following criteria are met:
•	The month and year (or year) of onset date is the same as or after the month and year (or year) of the first dose of the blinded study drug, and
•	For those who discontinued the blinded study drug permanently only: the month and year (or year) of onset date is the same as or before the month and year (or year) of the date of the minimum of the blinded study drug stop date + 3 days and the day prior to the first dose date of open-label study drug, if applicable, and
•	AE end date is as follows:
	The (complete) end date is on or after the first dose date of the blinded study drug, or
	The month and year (or year) of end date is the same or after the month and year (or year) of the first dose of the blinded study drug, or
	End date is completely missing
The event is treatment-emergent during the open-label phase if the following criteria are met:
•	The month and year (or year) of onset date is the same as or after the month and year (or year) of the first dose of the open-label study drug, and
•	For those who discontinued the blinded study drug permanently only: the month and year (or year) of onset date is the same as or before the month and year (or year) of the date of the last dose of the open-label study drug + 3 days, and
•	AE end date is as follows:
	The (complete) end date is on or after the first dose date of the open-label study drug, or
	The month and year (or year) of end date is the same or after the month and year (or year) of the first dose of the open-label study drug, or
	End date is completely missing
The event is post-treatment-emergent during the TFFU phase if the following criteria are met:
•	The month and year (or year) of onset date is the same as or after the month and year (or year) of the day after the last dose date + 3 days
Events with Completely Missing Onset Date
An AE with a completely missing onset date is defined as treatment-emergent AE during the double-blind phase if end date is as follows:
•	The (complete) end date is on or after the first dose date of the blinded study drug, or
•	The month and year (or year) of end date is the same or after the month and year (or year) of the first dose of the blinded study drug, or
•	End date is completely missing
An AE with a completely missing onset date is defined as treatment-emergent AE during the open-label phase if end date is as follows:
•	The (complete) end date is on or after the first dose date of the open-label study drug, or
•	The month and year (or year) of end date is the same or after the month and year (or year) of the first dose of the open-label study drug, or
•	End date is completely missing 
The event is post-treatment-emergent during the TFFU phase if the following criteria are met:
•	The (complete) end date is after the last dose date + 3 days, or
•	The month and year (or year) of end date is the same or after the month and year (or year) of the day after the last dose date + 3 days, or
•	End date is completely missing

**Treatment-Emergent Laboratory Abnormalities**
Treatment-emergent laboratory abnormalities occurring in the double-blind phase are defined as values that increase by at least 1 toxicity grade from baseline at any postbaseline visit up to and including the minimum of the blinded study drug stop date + 3 days and the first dose date of open-label study drug, if applicable, for those who discontinued blinded study drug permanently, or values that increase by at least 1 toxicity grade from baseline at any postbaseline visit for those who are still on double-blind study drug. If the relevant baseline laboratory value is missing, any laboratory abnormality of at least Grade 1 observed within the double-blind phase will be considered treatment emergent.
Treatment-emergent laboratory abnormalities occurring in the open-label phase are defined as values that increase by at least 1 toxicity grade from open-label baseline at any open-label postbaseline visit up to and including the last dose date of the open-label study drug + 3 days for those who discontinued open-label study drug permanently, or values that increase by at least 1 toxicity grade from open-label baseline at any open-label postbaseline visit for those who are still on open-label study drug. For the analyses of abnormalities occurring during open-label treatment, open-label baseline will be the last available record on or prior to Open-Label Study Day 1.
Post-treatment-emergent laboratory abnormalities occurring in the TFFU phase are defined as values that increase by at least 1 toxicity grade from TFFU baseline. For the analyses of abnormalities occurring during the TFFU phase, TFFU baseline will be the last available record on or prior to last dose date + 3 days.
Fasting glucose and nonfasting glucose (including glucose results without a known fasting status) are graded based on different grading scales as specified in the protocol. Treatment emergent and post-treatment-emergent laboratory abnormalities will be summarized for fasting glucose. Since nonfasting glucose was not assessed at baseline, the maximum postbaseline grade will be summarized.

**Comment by jwildfire** [comment]
My general thought is that we want to have the programming team cover as much of this as possible, and avoid doing too much duplication of standard AdAM derivations in gsm. I do recognize that won't be 100% doable (e.g. #10 & #12) ... Will just have to evaluate these on a case-by-case basis as they come up in our assessment derivations, I guess.

**Reply by gwu05** [reply]
Yeah, I think we'll just have to be clear on what assumptions we take to short-cut some of these outcomes definitions. So for treatment-emergent, essentially we just take what AEs are reporting after FDD and ignore the end date part of TEAE (or could even just take rows in AE more simply). Also, will be important to be able to also use ADaM when available as these will be defined and what the study teams will want to use.

**Comment by gwu05** [comment]
This should cover most of it: idea would be an ADSL lite for our assessment, then specific outcome information is separate data that needs to be provided/mapped.

ADSL lite: derivations are in []

SubjectID - dm? or subid?
SiteID - dm?  or subid?

[ScreenFlag]
[RandomizedFlag]
RandomizationDate - ixrsrand

TrtStartDate - min ex date
TrtEndDate - max ex dataset date
[TrtExposure] - TrtEndDate - TrtStartDate + 1 (days)

[TrtStatus] - On-going, Discontinued, Did not start?
TrtDiscontinuationReason - sdrgcomp* dataset

StudyStartDate - min visit dataset date
StudyEndDate - max visit dataset date
[TimeonStudy] - StudyEndDate - StudyStartDate + 1 (days) , if ongoing StudyEndDate is imputed as current date

[StudyStatus] - derived to be On-going, Completed, or Discontinued, and maybe "did not enter?"
StudyDiscontinuationReason - studcomp dataset
